packages/powerapps-lib/src/application-update/__tests__/batch-formation.spec.js

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable camelcase */
+import { openBatchRequest, createBatchRequestBody, findRequestSequence, createKeyObject } from '../batch-formation.js'
+import { model } from '../../model/sdds-model.js'
+import fs from 'fs'
+import path from 'path'
+
+export const src = {
+  applicant: {
+    lastname: 'Botham',
+    firstname: 'Ian',
+    email: 'Ian.botham@gmail.com',
+    phone: '876877666876',
+    address:
+        {
+          postcode: 'BS92LT',
+          addrline1: '1 The cottages',
+          addrline2: 'The Village',
+          addrline3: 'Taunton',
+          county: 'Somerset'
+        }
+  },
+  ecologist: {
+    firstname: 'Brian',
+    lastname: 'The-Ecologist',
+    email: 'brian.ecologist@gmail.com',
+    phone: '+44 837248649864',
+    address:
+      {
+        postcode: 'YT56 9UW',
+        addrline1: 'The University',
+        addrline2: 'University Rd.',
+        addrline3: 'Cambridge',
+        county: 'Cambridgeshire'
+      }
+  },
+  proposalDescription: 'move some newts across a road',
+  detailsOfConvictions: 'speeding fine 2008. 167mph.'
+}
+
+/**
+ * Content-Type:multipart/mixed;boundary=batch_499D1B
+ Accept:application/json
+ OData-MaxVersion: 4.0
+ OData-Version: 4.0
+ Prefer:return=representation
+ */
+describe('Batch query formation', () => {
+  it('forms a batch query set for a basic data subset', async () => {
+    const sequence = findRequestSequence({ sdds_applications: model.sdds_applications })
+    const batchId = openBatchRequest()
+
+    const clientUrl = 'https://sdds-dev.crm11.dynamics.com/api/data/v9.0'
```

Comment:
Did you mean to leave this `'https://sdds-dev.crm11.dynamics.com/api/data/v9.0'` here? I imagine this should be a dynamic string that comes from .env or the config object.

Reply:
Ignore me if this is not an actual URL and is just there for stubbing purposes!

Reply:
Yes - its just to match the example response in the tests